### ISSUE TITLE
Warnings about letterlike characters in identifiers

### ISSUE THREAD

**Issue by kenkangxgwe** [WolframResearch/codeinspector#8]
Hi, I wonder if there's a valid range for unicode characters that can appear in identifiers? Currently, Cyrillic letters, CJK characters, and other letterlike characters (like emoji 😂) will generate the `[UnexpectedLetterlikeCharacter] Unexpected letterlike character: \:xxxx` warning., e.g.
```mathematica
In[] := CodeInspect["\\:0438=1"]

Out[] := {
CodeInspector`InspectionObject[
 "UnexpectedLetterlikeCharacter", 
  "Unexpected letterlike character: ``\\:0438``.", "Warning", 
Association[
  CodeParser`Source -> {{1, 1}, {1, 7}}, ConfidenceLevel -> 0.8]]}
```

**Comment by bostick** [comment]
Yes, this is working as intended.

For example, a few weeks ago this caught a real bug in the Mathematica startup code when this code was mistakenly added:
```
аnimationRepetitions = foo[]
```

Can you see the problem? It is subtle. (I deliberately did not turn on mathematica coloring)

I do agree that there might be value in allowing some ranges of characters but I'm afraid it would be a slippery slope and I'm not sure what that design would look like.

In the meanwhile, feel free to programmatically filter the output of `CodeInspect`.

**Comment by szhorvat** [comment]
Lints can always be turned off. I find it useful when it reports _more_ rather than less, and I simply turn off whatever is not relevant to my own use case. (In fact, for every package on which I use this tool, I have a list of lints to turn off.)

Just don't fall into the trap of trying to change your code just to shut up the linter. It's a tool meant to help you find issues, not an overlord to satisfy 😄

**Comment by kenkangxgwe** [comment]
I would turn the whole `UnexpectedLetterlikeCharacter` off, but seems that it shares the same name when the characters fall into `Utils::isMBStrange`. The only difference is that the confidence is higher for the latter case. I wonder if it is true that for `isMBStrange` characters, a `Unexpected character` warning will also be thrown? So that I don't need to worry that supressing `UnexpectedLetterlikeCharacter` will ignore the non-characters.

**Comment by bostick** [comment]
The idea is that `UnexpectedLetterlikeCharacter` lints are at the "token" level and `UnexpectedCharacter` lints are at the "character" level.

Right now, any character that gives `UnexpectedLetterlikeCharacter` will also give `UnexpectedCharacter`.

`isMBStrange` is for the "character" level and `isMBStrangeLetterlike` is for the token level.

**Comment by kenkangxgwe** [comment]
Thanks for the explanation. Closing this issue.
